examples/algorithm_progression/xarray_sweep.py

```diff
@@ -0,0 +1,87 @@
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+from algorithm_sweep import AlgorithmSweep
+
+import large_image
+
+
+class AlgorithmSweepXArray(AlgorithmSweep):
+    def create_ds(self):
+        dim_sizes = dict(
+            **{k: len(v) for k, v in self.input_params.items()},
+            **{
+                'y': self.source_metadata['sizeY'],
+                'x': self.source_metadata['sizeX'],
+                'c': 4,
+            },
+        )
+
+        ds = xr.Dataset(
+            {
+                'results': (
```

Comment:
We'd ideally like to NOT allocate all the space up front.  Can we create this with expandable dimensions?

Reply:
Changed in [980bc50](https://github.com/girder/large_image/pull/1077/commits/980bc50d5ca759524577d1656b72dfa62f6aab96)